specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json

```diff
@@ -0,0 +1,5736 @@
+            "Script",
+            "Location"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "kind",
+            "modelAsString": true
+          }
+        }
+      },
+      "discriminator": "kind"
+    },
+    "ConstraintList": {
+      "description": "List response for get Constraints",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "The Url of next result page.",
+          "type": "string"
+        },
+        "value": {
+          "description": "Collection of items of type DataTransferObjects.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Constraint"
+          }
+        }
+      }
+    },
+    "ConsumerInvitation": {
+      "description": "A consumer Invitation data transfer object.",
+      "required": [
+        "properties"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyDto"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ConsumerInvitationProperties",
+          "description": "Properties on the account",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ConsumerInvitationList": {
+      "description": "List response for get InvitationList",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "The Url of next result page.",
+          "type": "string"
+        },
+        "value": {
+          "description": "Collection of items of type DataTransferObjects.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConsumerInvitation"
+          }
+        }
+      }
+    },
+    "ConsumerInvitationProperties": {
+      "description": "Properties of consumer invitation",
+      "required": [
+        "invitationId"
+      ],
+      "type": "object",
+      "properties": {
+        "description": {
+          "description": "Description shared when the invitation was created",
+          "type": "string",
+          "readOnly": true
+        },
+        "invitationId": {
+          "description": "Unique id of the invitation.",
+          "type": "string"
+        },
+        "invitationStatus": {
+          "description": "The status of the invitation.",
+          "enum": [
+            "Pending",
+            "Accepted",
+            "Rejected",
+            "Withdrawn"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "invitationStatus",
+            "modelAsString": true
+          }
+        },
+        "location": {
+          "description": "invitation location",
+          "type": "string",
+          "readOnly": true
+        },
+        "proposalName": {
```

Comment:
Is this a proposal from another tenant?  Should this use a full resource ID so it is uniquely identifiable (name won't be unless you are enforcing global uniqueness on the backend).

Reply:
It is from another tenant and we aren't enforcing name uniqueness  - but I prefer to leave it just name because it's not used by the system for identification, it's just a hint for the user, and receiving user will use the invitation to create a new proposal with a new name (and different id) - in general we avoided sharing Azure Ids between users in different tenants.